### PR TITLE
Add support for multiple controlnet

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

- [x] Export of multiple controlnets

```bash
optimum-cli export neuron --inline-weights-neff --model jyoung105/stable-diffusion-v1-5 --task stable-diffusion --auto_cast matmul --auto_cast_type bf16 --batch_size 1 --num_images_per_prompt 1 --controlnet_ids lllyasviel/control_v11p_sd15_openpose lllyasviel/control_v11f1p_sd15_depth --height 512 --width 512 sd15-512x512-bf16-openpose-depth
```
- [x] Support for multiple ControlNets for stable diffusion

```python
import numpy as np
import torch
from PIL import Image

from controlnet_aux import OpenposeDetector
from transformers import pipeline
from diffusers import UniPCMultistepScheduler
from diffusers.utils import load_image
from optimum.neuron import NeuronStableDiffusionControlNetPipeline


# OpenPose+Depth ControlNet
model_id = "sd15-512x512-bf16-openpose-depth"

# Load ControlNet images

# 1. openpose
image_0 = load_image("https://huggingface.co/lllyasviel/control_v11p_sd15_openpose/resolve/main/images/input.png")
processor = OpenposeDetector.from_pretrained('lllyasviel/ControlNet')
openpose_image = processor(image_0)

# 2. depth
image_1 = load_image("https://huggingface.co/lllyasviel/control_v11p_sd15_depth/resolve/main/images/input.png")
depth_estimator = pipeline('depth-estimation')
image_1 = depth_estimator(image_1)['depth']
image_1 = np.array(image_1)
image_1 = image_1[:, :, None]
image_1 = np.concatenate([image_1, image_1, image_1], axis=2)
depth_image = Image.fromarray(image_1)

images = [openpose_image.resize((512, 512)), depth_image.resize((512, 512))]

# Inference
pipe = NeuronStableDiffusionControlNetPipeline.from_pretrained(model_id)
pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)
prompt = "a giant in a fantasy landscape, best quality"
negative_prompt = "monochrome, lowres, bad anatomy, worst quality, low quality"

image = pipe(prompt=prompt, image=images).images[0]
image.save('inf2_out.png')
```

- [x] Support for multiple ControlNets for SDXL

* Export

```bash
optimum-cli export neuron -m stabilityai/stable-diffusion-xl-base-1.0 --task stable-diffusion-xl --batch_size 1 --height 1024 --width 1024 --controlnet_ids diffusers/controlnet-canny-sdxl-1.0-small thibaud/controlnet-openpose-sdxl-1.0 --num_images_per_prompt 1 sdxl_neuron_canny_openpose/
```

* Inference
Theoretically yes, but easily get oom without parallelism which will be supported with future NxD integration for inference. 

- [x] Tests
- [x] Documentation

- [x] Some changes to fix CIs

It seems that the CIs failed after we bump the transformers/optimum versions:

* Deprecating `use_auth_token` since the CIs failed with recent changes in Optimum: https://github.com/huggingface/optimum/pull/1837
* Fix generation mixing following: https://github.com/huggingface/transformers/pull/31980

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by mludvig** [comment]
Hi @JingyaHuang that's great, it works! Thanks a lot for providing the fix 🎉

File: optimum/neuron/modeling_decoder.py [hunk 1]
```diff
@@ -467,24 +466,13 @@ def push_to_hub(
...
+        token: Union[bool, str] = True,
         endpoint: Optional[str] = None,
     ) -> str:
-        if isinstance(use_auth_token, str):
-            huggingface_token = use_auth_token
-        elif use_auth_token:
-            huggingface_token = get_token()
-        else:
-            raise ValueError("You need to provide `use_auth_token` to be able to push to the hub")
         api = HfApi(endpoint=endpoint)
```
Comment (dacorvo) [inline optimum/neuron/modeling_decoder.py:472]: nit: I actually think we should pass the token here, so that it can be omitted later.

cc @Wauplin

**Review by dacorvo — APPROVED** [review]
LGTM, thank you for the pull-request !

File: docs/source/inference_tutorials/stable_diffusion.mdx [hunk 2]
```diff
@@ -635,6 +635,68 @@ compare.save("compare.png")
...
+depth_estimator = pipeline('depth-estimation')
+image = depth_estimator(image)['depth']
+image = np.array(image)
+image = image[:, :, None]
+image = np.concatenate([image, image, image], axis=2)
+depth_image = Image.fromarray(image)
+
+images = [openpose_image.resize((512, 512)), depth_image.resize((512, 512))]
+
+# Inference
```
Comment (michaelbenayoun) [inline docs/source/inference_tutorials/stable_diffusion.mdx:683]:
```suggestion
# 3. inference
```

Comment (michaelbenayoun) [reply docs/source/inference_tutorials/stable_diffusion.mdx:683]: nit just to follow the same style as the previous comments.

File: optimum/neuron/modeling_diffusion.py [hunk 3]
```diff
@@ -758,9 +758,9 @@ def _export(
                 configuration files of compatible classes.
             unet_id (`Optional[Union[str, Path]]`, defaults to `None`):
                 A string or a path point to the U-NET model to replace the one in the original pipeline.
-            use_auth_token (`Optional[Union[bool, str]]`, defaults to `None`):
+            token (`Optional[Union[bool,str]]`, defaults to `None`):
```
Comment (michaelbenayoun) [inline optimum/neuron/modeling_diffusion.py:761]:
```suggestion
            token (`Optional[Union[bool, str]]`, defaults to `None`):
```

**Review by michaelbenayoun — APPROVED** [review]
LGTM

File: optimum/neuron/modeling_decoder.py (hunk 1 above)
Comment (JingyaHuang) [reply optimum/neuron/modeling_decoder.py:472]: that sounds good, I just did it the way it was to pass the CIs, I will leave it to you if you want to change! (don't want to wait for CIs again hhh

File: docs/source/inference_tutorials/stable_diffusion.mdx (hunk 2 above)
Comment (JingyaHuang) [reply docs/source/inference_tutorials/stable_diffusion.mdx:683]: good catch!

**Pull request merged**